eclass/toolchain-funcs.eclass

```diff
@@ -421,6 +421,49 @@ tc-has-openmp() {
 	return ${ret}
 }
 
+# @FUNCTION: tc-ensure-openmp
+# @DESCRIPTION:
+# If 'openmp' is in IUSE and enabled OR if 'openmp' is not in IUSE, test
+# whether the currently active compiler supports OpenMP and die if it does
+# not. If it dies, it gives a detailed error message to the user, warning
+# them of the failure. This function should be used in pkg_setup (and
+# optionally in pkg_pretend).
+tc-ensure-openmp() {
+	case "${EAPI:-0}" in
+		[0-5])
+			die "Unsupported EAPI=${EAPI:-0} (too old) for tc-ensure-openmp"
+			;;
+		*)
+			;;
+	esac
+
+	if [[ ${MERGE_TYPE} != binary ]]; then
+		local openmp_requested
+		local die_message
+		if in_iuse openmp && use openmp; then
```

Comment:
I needed to read this three times before WTF turned into lightbulb. At least add an explanatory comment.